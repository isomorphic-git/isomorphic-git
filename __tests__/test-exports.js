/* eslint-env node, browser, jasmine */
// @ts-ignore
const snapshots = require('./__snapshots__/test-exports.js.snap')
const registerSnapshots = require('./__helpers__/jasmine-snapshots')

const git = require('isomorphic-git')

describe('exports', () => {
  beforeAll(() => {
    registerSnapshots(snapshots)
  })
  it('exposes only the intended API functions', async () => {
    const names = Object.keys(git)
    expect(names.sort()).toMatchSnapshot()
  })
})
