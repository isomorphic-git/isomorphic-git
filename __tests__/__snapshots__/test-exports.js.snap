// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exports exposes only the intended API functions 1`] = `
Array [
  "E",
  "STAGE",
  "TREE",
  "WORKDIR",
  "add",
  "addRemote",
  "annotatedTag",
  "branch",
  "checkout",
  "clone",
  "commit",
  "config",
  "cores",
  "currentBranch",
  "deleteBranch",
  "deleteRef",
  "deleteRemote",
  "deleteTag",
  "expandOid",
  "expandRef",
  "fetch",
  "findMergeBase",
  "findRoot",
  "getRemoteInfo",
  "hashBlob",
  "indexPack",
  "init",
  "isDescendent",
  "listBranches",
  "listFiles",
  "listRemotes",
  "listTags",
  "log",
  "merge",
  "packObjects",
  "plugins",
  "pull",
  "push",
  "readObject",
  "remove",
  "resetIndex",
  "resolveRef",
  "sign",
  "status",
  "statusMatrix",
  "tag",
  "utils",
  "verify",
  "version",
  "walkBeta1",
  "walkBeta2",
  "writeObject",
  "writeRef",
]
`;
